### PR TITLE
Fixes parallel runs with solve_welleq_initially

### PR DESCRIPTION
**Pull request by blattms**
If this option was set there were some branches in
the code that did depend on the local number of wells
but should depend on the number of wells in the reservoir
no matter on which process they are stored.

With this commit we introduce BlackOilModelBase::localWellsActive()
which only takes local wells into account. The function now
BlackOilModelBase::wellsActive() considers all active wells in the
reservoir.

### CODE REVIEW

**Comment by alfbr** [comment]
Fantastic, thanks for the speedy progress! Crossing fingers for progress on scaling :-)

**Comment by blattms** [comment]
Unforntunately, I still have some issues with Norne at a later simulation step.

File: opm/autodiff/BlackoilModelBase_impl.hpp [hunk 1]
```diff
@@ -180,15 +180,23 @@ namespace detail {
...
-                    boost::any_cast<const ParallelISTLInformation&>(linsolver_.parallelInformation());
+        if ( linsolver_.parallelInformation().type() == typeid(ParallelISTLInformation) )
+        {
+            const ParallelISTLInformation& info =
+                boost::any_cast<const ParallelISTLInformation&>(linsolver_.parallelInformation());
+            if ( terminal_output_ ) {
                 // Only rank 0 does print to std::cout if terminal_output is enabled
                 terminal_output_ = (info.communicator().rank()==0);
             }
+            int global_number_of_wells = wells_->number_of_wells;
```
Comment (atgeirr) [inline opm/autodiff/BlackoilModelBase_impl.hpp]: At this point, it is the local, not global number of wells, so this variable should be renamed, and a separate (global) used for the sum.

**Comment by atgeirr** [comment]
This looks fine, but made me wonder: where is wellsActive() still used (i.e. not changed to localWellsActive())?

File: opm/autodiff/BlackoilModelBase_impl.hpp (hunk 1 above)
Comment (blattms) [inline opm/autodiff/BlackoilModelBase_impl.hpp]: Fixed.
-- 
Join us at the DUNE User Meeting Sept. 28-29, 2015, Heidelberg, Germany
http://users.dune-project.org/projects/dune-user-meeting-2015/wiki

**Comment by blattms** [comment]
There is at least one instance of wellsActive(). But I am not sure
whether that is not a bug. I need to recheck. Please do not merge yet.
-- 
Join us at the DUNE User Meeting Sept. 28-29, 2015, Heidelberg, Germany
http://users.dune-project.org/projects/dune-user-meeting-2015/wiki

**Comment by blattms** [comment]
That one is needed as solveWellEq contains some global reductions in the convergence check that need all processes or will otherwise deadlock.

**Comment by atgeirr** [comment]
So this is merge-ready?

**Comment by blattms** [comment]
I guess we can merge this. With interleaved norne I am experiences
wells being distributed. But that should not be correlated to this.

In any case that should be no changes for runs with one process.

**Pull request merged**